### ISSUE TITLE
AttributeError: module 'tables' has no attribute 'exceptions'

### ISSUE THREAD

**Issue by CorinneAKelly** [DeepLabCut/DeepLabCut#1398]
I am on a Windows 10 PC, DLC 2.2rc3 in an anaconda env.

I am trying to use the refine tracklets tab in the GUI. Originally, I was getting the error that "tables" didn't exist, and to pip install it. However, when I tried that it simply said it was already there. Someone on another thread mentioned that they had to uninstall pytables then pip install tables, so I tried this. Now I am getting a new error that I'm not sure how to resolve.

<details><summary>Code output</summary><p>

NoSuchNodeError                           Traceback (most recent call last)
~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in get_node(self, key)
   1502         try:
-> 1503             node = self._handle.get_node(self.root, key)
   1504         except _table_mod.exceptions.NoSuchNodeError:

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\tables\file.py in get_node(self, where, name, classname)
   1634             nodepath = join_path(basepath, name or '') or '/'
-> 1635             node = where._v_file._get_node(nodepath)
   1636         elif isinstance(where, (str, numpy.str_)):

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\tables\file.py in _get_node(self, nodepath)
   1589
-> 1590         node = self._node_manager.get_node(nodepath)
   1591         assert node is not None, "unable to instantiate node ``%s``" % nodepath

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\tables\file.py in get_node(self, key)
    431         if self.node_factory:
--> 432             node = self.node_factory(key)
    433             self.cache_node(node, key)

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\tables\group.py in _g_load_child(self, childname)
   1177         # Is the node a group or a leaf?
-> 1178         node_type = self._g_check_has_child(childname)
   1179

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\tables\group.py in _g_check_has_child(self, name)
    392         if node_type == "NoSuchNode":
--> 393             raise NoSuchNodeError(
    394                 "group ``%s`` does not have a child named ``%s``"

NoSuchNodeError: group ``/`` does not have a child named ``/tracks``

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\gui\refine_tracklets.py in create_tracks(self, event)
    313
    314     def create_tracks(self, event):
--> 315         deeplabcut.stitch_tracklets(
    316             self.config,
    317             self.datafile,

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py in stitch_tracklets(config_path, pickle_file, n_tracks, min_length, split_tracklets, prestitch_residuals, max_gap, weight_func, output_name)
   1048     stitcher.build_graph(max_gap=max_gap, weight_func=weight_func)
   1049     stitcher.stitch()
-> 1050     stitcher.write_tracks(output_name, animal_names)
   1051     return stitcher

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py in write_tracks(self, output_name, animal_names)
    867         if not output_name:
    868             output_name = self.filename.replace("pickle", "h5")
--> 869         df.to_hdf(output_name, "tracks", format="table", mode="w")
    870
    871     @staticmethod

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\core\generic.py in to_hdf(self, path_or_buf, key, mode, complevel, complib, append, format, index, min_itemsize, nan_rep, dropna, data_columns, errors, encoding)
   2698         from pandas.io import pytables
   2699
-> 2700         pytables.to_hdf(
   2701             path_or_buf,
   2702             key,

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in to_hdf(path_or_buf, key, value, mode, complevel, complib, append, format, index, min_itemsize, nan_rep, dropna, data_columns, errors, encoding)
    312             path_or_buf, mode=mode, complevel=complevel, complib=complib
    313         ) as store:
--> 314             f(store)
    315     else:
    316         f(path_or_buf)

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in <lambda>(store)
    294     else:
    295         # NB: dropna is not passed to `put`
--> 296         f = lambda store: store.put(
    297             key,
    298             value,

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in put(self, key, value, format, index, append, complib, complevel, min_itemsize, nan_rep, data_columns, encoding, errors, track_times, dropna)
   1120             format = get_option("io.hdf.default_format") or "fixed"
   1121         format = self._validate_format(format)
-> 1122         self._write_to_group(
   1123             key,
   1124             value,

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in _write_to_group(self, key, value, format, axes, index, append, complib, complevel, fletcher32, min_itemsize, chunksize, expectedrows, dropna, nan_rep, data_columns, encoding, errors, track_times)
   1753             return
   1754
-> 1755         group = self._identify_group(key, append)
   1756
   1757         s = self._create_storer(group, format, value, encoding=encoding, errors=errors)

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in _identify_group(self, key, append)
   1796     def _identify_group(self, key: str, append: bool) -> Node:
   1797         """Identify HDF5 group based on key, delete/create group if needed."""
-> 1798         group = self.get_node(key)
   1799
   1800         # we make this assertion for mypy; the get_node call will already

~\Anaconda3\envs\DEEPLABCUT\lib\site-packages\pandas\io\pytables.py in get_node(self, key)
   1502         try:
   1503             node = self._handle.get_node(self.root, key)
-> 1504         except _table_mod.exceptions.NoSuchNodeError:
   1505             return None
   1506

AttributeError: module 'tables' has no attribute 'exceptions'

</p></details>

**How to Reproduce the problem**
1. Go to refine tracklets tab
2. Click on step 1: create tracks

**Comment by backyardbiomech** [comment]
Uninstalling and re-installing potables can cause some conflicts in your env. It worked for me once, but I just tested the method again and it failed. 
I would create a new deeplabcut env from scratch (removing the old one is up to you).

To remove (assuming the env name is DEEPLABCUT):

`conda activate base`
`conda env remove -n DEEPLABCUT`

Go into your downloads folder and delete and previously downloaded DEEPLABCUT.yaml files. Download the DEEPLABCUT.yaml file from [here](http://www.mackenziemathislab.org/s/DEEPLABCUT.yaml).  This is the updated one from the blog. I have tested this one multiple times and it works.

Assuming this shows up in your Downloads folder, and let's say you don't want to remove your env, so you need a new name for this new env, like DLCrc3 (or whatever you want):

`cd C:\Users\yourusername\Downloads`
`conda env create -n DLCrc3 -f DEEPLABCUT.yaml`

If you have a GPU, you might need to separately install CUDA with:

`conda install -c conda-forge cudnn`

**Comment by CorinneAKelly** [comment]
Thank you so much for the help.

I followed your instructions for removing the old env. I also updated cudnn again just in case and that seemed to go fine. However, after downloading the new yaml file (I moved it to the folder it was previously located for ease of access/organization) and attempting to create a new env I get the following error:
```
(base) C:\Users\ckelly\Documents\Git repositories\DeepLabCut\conda-environments>conda env create -n DLC -f DEEPLABCUT.yaml

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda\exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\cli\main.py", line 80, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\cli\main_create.py", line 88, in execute
        spec = specs.detect(name=name, filename=get_filename(args.file), directory=os.getcwd())
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\specs\__init__.py", line 43, in detect
        if spec.can_handle():
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\specs\yaml_file.py", line 18, in can_handle
        self._environment = env.from_file(self.filename)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\env.py", line 166, in from_file
        return from_yaml(yamlstr, filename=filename)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda_env\env.py", line 143, in from_yaml
        data = yaml_safe_load(yamlstr)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\conda\common\serialize.py", line 67, in yaml_safe_load
        return yaml.safe_load(string, version="1.2")
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\main.py", line 980, in safe_load
        return load(stream, SafeLoader, version)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\main.py", line 935, in load
        return loader._constructor.get_single_data()
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\constructor.py", line 109, in get_single_data
        node = self.composer.get_single_node()
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 78, in get_single_node
        document = self.compose_document()
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 101, in compose_document
        node = self.compose_node(None, None)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 138, in compose_node
        node = self.compose_mapping_node(anchor)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 218, in compose_mapping_node
        item_value = self.compose_node(node, item_key)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 136, in compose_node
        node = self.compose_sequence_node(anchor)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 180, in compose_sequence_node
        node.value.append(self.compose_node(node, index))
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 138, in compose_node
        node = self.compose_mapping_node(anchor)
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\composer.py", line 211, in compose_mapping_node
        while not self.parser.check_event(MappingEndEvent):
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\parser.py", line 141, in check_event
        self.current_event = self.state()
      File "C:\Users\ckelly\Anaconda3\lib\site-packages\ruamel_yaml\parser.py", line 574, in parse_block_mapping_key
        raise ParserError(
    ruamel_yaml.parser.ParserError: while parsing a block mapping
      in "<unicode string>", line 26, column 5:
          - pip:
            ^ (line: 26)
    expected <block end>, but found '<scalar>'
      in "<unicode string>", line 27, column 24:
            - 'deeplabcut[gui]'==2.2rc3
                               ^ (line: 27)

`$ C:\Users\ckelly\Anaconda3\Scripts\conda-env-script.py create -n DLC -f DEEPLABCUT.yaml`

  environment variables:
                 CIO_TEST=<not set>
  CONDA_AUTO_UPDATE_CONDA=false
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=C:\Users\ckelly\Anaconda3\condabin\..\Scripts\conda.exe
               CONDA_EXES="C:\Users\ckelly\Anaconda3\condabin\..\Scripts\conda.exe"
             CONDA_PREFIX=C:\Users\ckelly\Anaconda3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=C:\Users\ckelly\Anaconda3\python.exe
               CONDA_ROOT=C:\Users\ckelly\Anaconda3
              CONDA_SHLVL=1
                CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0
           CURL_CA_BUNDLE=<not set>
                 HOMEPATH=\Users\ckelly
          NVTOOLSEXT_PATH=C:\Program Files\NVIDIA Corporation\NvToolsExt\
                     PATH=C:\Users\ckelly\Anaconda3;C:\Users\ckelly\Anaconda3\Library\mingw-w64\
                          bin;C:\Users\ckelly\Anaconda3\Library\usr\bin;C:\Users\ckelly\Anaconda
                          3\Library\bin;C:\Users\ckelly\Anaconda3\Scripts;C:\Users\ckelly\Anacon
                          da3\bin;C:\Users\ckelly\Anaconda3;C:\Users\ckelly\Anaconda3\Library\mi
                          ngw-w64\bin;C:\Users\ckelly\Anaconda3\Library\usr\bin;C:\Users\ckelly\
                          Anaconda3\Library\bin;C:\Users\ckelly\Anaconda3\Scripts;C:\Users\ckell
                          y\Anaconda3\bin;C:\Users\ckelly\Anaconda3\condabin;C:\Program
                          Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin;C:\Program
                          Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\libnvvp;C:\WINDOWS\syste
                          m32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPow
                          erShell\v1.0;C:\WINDOWS\System32\OpenSSH;C:\Program Files (x86)\NVIDIA
                          Corporation\PhysX\Common;C:\Program Files\Git\cmd;C:\Program
                          Files\NVIDIA Corporation\NVIDIA
                          NvDLISR;C:\Users\ckelly\AppData\Local\Microsoft\WindowsApps
             PSMODULEPATH=C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\Windows
                          PowerShell\v1.0\Modules;C:\Program Files\Thycotic\Powershell\
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : C:\Users\ckelly\Anaconda3
            shell level : 1
       user config file : C:\Users\ckelly\.condarc
 populated config files : C:\Users\ckelly\.condarc
          conda version : 4.10.3
    conda-build version : 3.20.5
         python version : 3.8.5.final.0
       virtual packages : __cuda=11.4=0
                          __win=0=0
                          __archspec=1=x86_64
       base environment : C:\Users\ckelly\Anaconda3  (writable)
      conda av data dir : C:\Users\ckelly\Anaconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\ckelly\Anaconda3\pkgs
                          C:\Users\ckelly\.conda\pkgs
                          C:\Users\ckelly\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\ckelly\Anaconda3\envs
                          C:\Users\ckelly\.conda\envs
                          C:\Users\ckelly\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.10.3 requests/2.24.0 CPython/3.8.5 Windows/10 Windows/10.0.19041
          administrator : True
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```

Any idea what might have gone wrong here?

**Comment by backyardbiomech** [comment]
Open the yaml file in a text editor. Remove the single quotes around `deeplabcut[gui]`. Save. Try again.

**Comment by CorinneAKelly** [comment]
This worked to create my environment! Now when importing deeplabcut, I get this error:

```
In [1]: import deeplabcut
2021-07-11 12:27:43.457800: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2021-07-11 12:27:43.457889: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
```

I do have a GPU set up, so should I be worried about this?

**Comment by backyardbiomech** [comment]
Make sure CUDA 11 is installed.

run `nvcc -V` in the env to see what CUDA is installed. If you didn't update your GPU driver, you might need to do that to get a version that is compatible with CUDA 11 and TensorFlow 2.5. The main installation page has links for installing compatible versions of everything. 

I had to update my GPU driver, then CUDA, then deeplabcut, then cudnn,  when I upgraded.
